sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryRelation.scala

```diff
@@ -277,6 +277,9 @@ object InMemoryRelation {
     ser.get
   }
 
+  /* Visible for testing */
+  private[spark] def clearSerializer(): Unit = synchronized { ser = None }
```

Comment:
Nice, @dongjoon-hyun. What about just fixing `CachedBatchSerializerSuite` not to extend `SharedSparkSessionBase`? For example, like `ExecutorSideSQLConfSuite` or `SparkSessionExtensionSuite`. I think that would be simpler and a more isolated fix.

Reply:
Ya. I thought that way first, but this is more general way because SPARK-32274 make SQL cache serialization pluggable. We may have another test suite in the future.

Reply:
Also, the test resource clean-up had better be centralized at `SharedSparkSession`.

Reply:
BTW, I'm also open to your idea. Let's see the original author and committer opinion. Thanks.

Reply:
Does it really help? `InMemoryRelation.ser` doesn't belong to any session and is global.

I think a simpler fix is to clear it in `CachedBatchSerializerSuite.beforeAll` and `afterAll`.

Reply:
A more complicated fix is to put the serializer instance in `SharedState`. This brings one benefit: users can stop the current spark context, create a new one with a different cache plugin. But I'm not sure if this is a useful feature.

Reply:
Yes. Since this is a new feature and new test coverage, I'm also not sure what is the best here.

Reply:
For this question, yes. The root cause is that InMemoryRelation.ser is a kind of singleton. Since the new configuration is static conf, this will match with the semantic of `InMemoryRelation.ser`. So, the problem is the testing.
> Does it really help? InMemoryRelation.ser doesn't belong to any session and is global.

Reply:
Let me create another PR for @HyukjinKwon or @cloud-fan to compare.

Reply:
First of all, @HyukjinKwon 's idea is not helpful because `InMemoryRelation.ser` is a singleton.
> What about just fixing CachedBatchSerializerSuite not to extend SharedSparkSessionBase?

I'm moving to @cloud-fan 's proposal.

Reply:
Thanks @dongjoon-hyun!

Reply:
I like @cloud-fan proposal as it will make testing easier as well.